x/metadata/types/msg.go

```diff
@@ -672,3 +679,141 @@ func (msg MsgDeleteRecordSpecificationRequest) ValidateBasic() error {
 	}
 	return nil
 }
+
+// --------------------------- OSLocator------------------------------------------
+
+// NewMsgBindOSLocatorRequest creates a new msg instance
+func NewMsgBindOSLocatorRequest(obj ObjectStoreLocator) *MsgBindOSLocatorRequest {
+	return &MsgBindOSLocatorRequest{
+		Locator: obj,
+	}
+}
+
+func (msg MsgBindOSLocatorRequest) Route() string {
+	return ModuleName
+}
+
+func (msg MsgBindOSLocatorRequest) Type() string {
+	return TypeMsgBindOSLocatorRequest
+}
+
+func (msg MsgBindOSLocatorRequest) ValidateBasic() error {
+	err := ValidateOSLocatorObj(msg.Locator.Owner, msg.Locator.LocatorUri)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (msg MsgBindOSLocatorRequest) GetSignBytes() []byte {
+	return sdk.MustSortJSON(ModuleCdc.MustMarshalJSON(&msg))
+}
+
+func (msg MsgBindOSLocatorRequest) GetSigners() []sdk.AccAddress {
+	addr, err := sdk.AccAddressFromBech32(msg.Locator.Owner)
+	if err != nil {
+		panic(err)
+	}
+	return []sdk.AccAddress{addr}
+}
+
+// ---- Delete OS locator ------
+func NewMsgDeleteOSLocatorRequest(obj ObjectStoreLocator) *MsgDeleteOSLocatorRequest {
+	return &MsgDeleteOSLocatorRequest{
+		Locator: obj,
+	}
+}
+func (msg MsgDeleteOSLocatorRequest) Route() string {
+	return ModuleName
+}
+
+func (msg MsgDeleteOSLocatorRequest) Type() string {
+	return TypeMsgDeleteOSLocatorRequest
+}
+
+func (msg MsgDeleteOSLocatorRequest) ValidateBasic() error {
```

Comment:
lots of code coverage warnings for this file ... these could be mitigated pretty quickly with a couple more basic tests added to the `msg_test.go` file

Reply:
did add some for this too.. looks green'ish  on new code